rasa/server.py

```diff
@@ -951,19 +1072,61 @@ async def evaluate_stories(request: Request) -> HTTPResponse:
 
     @app.post("/model/test/intents")
     @requires_auth(app, auth_token)
-    async def evaluate_intents(request: Request) -> HTTPResponse:
+    @async_if_callback_url
+    @heavy_computation
+    @inject_temp_dir
+    async def evaluate_intents(
+        request: Request, temporary_directory: Path
+    ) -> HTTPResponse:
         """Evaluate intents against a Rasa model."""
         validate_request_body(
             request,
             "You must provide some nlu data in the request body in order to "
             "evaluate your model.",
         )
 
-        test_data = _test_data_file_from_payload(request)
+        is_yaml_payload = request.headers.get("Content-type") == YAML_CONTENT_TYPE
+        config_file = None
+        if is_yaml_payload:
+            payload = _training_payload_from_yaml(request, temporary_directory)
+            config_file = payload.get("config")
+            test_data = payload.get("training_files")
+        else:
+            test_data = _test_data_file_from_payload(request, temporary_directory)
 
+        cross_validation_folds = request.args.get("cross_validation_folds")
+
+        if cross_validation_folds and is_yaml_payload:
+            test_coroutine = _cross_validate(
+                test_data, config_file, int(cross_validation_folds)
+            )
+        elif cross_validation_folds:
+            raise ErrorResponse(
+                HTTPStatus.BAD_REQUEST,
+                "TestingError",
+                "Cross-validation is only supported for YAML data.",
+            )
+        else:
```

Comment:
Since all these `if`, `elif` and `else`s are referring to 2 flags (`cross_validation_folds`, and `is_yaml_payload`) a nested `if` might be clearer?